### PR TITLE
Adds the 'Submit to QuickStatements' button

### PR DESCRIPTION
**Pull request by egonw**
Implements https://github.com/fnielsen/scholia/issues/1458

### CODE REVIEW

**Comment by Daniel-Mietchen** [comment]
The code looks good, and I just created https://www.wikidata.org/wiki/Q107308161 while testing it.

File: scholia/app/templates/404_doi.html [hunk 1]
```diff
@@ -1,13 +1,40 @@
...
+try {
+  let example = new c.Cite([ '{{ doi }}' ])
+  let output = example.format('quickstatements')
+  $( '#qs' ).append( output );
+  output = encodeURI(
+    output.replaceAll('\t', '|')
+          .replaceAll('\n', '||'))
+          .replaceAll('/','%2F')
+  htmlOutput = "<a href=\"https://quickstatements.toolforge.org/#/v1=" + output + "\">\n" +
+               "  <button class=\"btn btn-primary\">Submit to Quickstatements ↗</button>\n" +
```
Comment (Daniel-Mietchen) [inline scholia/app/templates/404_doi.html:18]: ```Quickstatements``` should be ```QuickStatements```.

File: scholia/app/templates/404_doi.html [hunk 2]
```diff
@@ -1,13 +1,40 @@
...
+    console.log(error)
+  }
+}
+</script>
+
+{% endblock %}
+
 {% block page_content %}
 
 There was no work found with DOI <a href="https://doi.org/{{ doi }}">{{ doi }}</a>.
```
Comment (Daniel-Mietchen) [inline scholia/app/templates/404_doi.html:34]: I suggest to change this to 
```We found no work in Wikidata with the DOI <a href="https://doi.org/{{ doi }}">{{ doi }}</a>. If you would like to create the item for this work, you can use the <a href="https://www.wikidata.org/wiki/Help:QuickStatements">QuickStatements</a> tool by clicking the "Submit to QuickStatements" button (this requires authentication via <a href="https://www.mediawiki.org/wiki/Help:OAuth">OAuth 2.0</a>).```

**Review by Daniel-Mietchen — CHANGES_REQUESTED** [review]
Works fine, but presentation could be improved.

File: scholia/app/templates/404_doi.html (hunk 1 above)
Comment (egonw) [reply scholia/app/templates/404_doi.html:18]: Good catch. Sorry I overlooked that.

File: scholia/app/templates/404_doi.html (hunk 2 above)
Comment (egonw) [reply scholia/app/templates/404_doi.html:34]: Sounds good to me.

**Comment by fnielsen** [comment]
Is there a need of `urlencode`?

**Comment by egonw** [comment]
> Is there a need of `urlencode`?

I copied that from the arxiv example and assumed so.

**Comment by fnielsen** [comment]
arXiv has \d+.\d+ pattern I think. I don't know if that is checked. Can there be '"' in a DOI? '<' and '>' might not be a problem.

**Comment by egonw** [comment]
I've seen many interesting characters in DOIs. The regex in identifiers.org is `^(doi\:)?\d{2}\.\d{4}.*$`

**Comment by fnielsen** [comment]
So I suppose we need an urlencoding?

**Comment by egonw** [comment]
Yes, I think https://github.com/WDscholia/scholia/pull/1461/files#diff-89c1152879b643f0f5eb07072531c16873bb0581597c328b6efa86fddbd72c79R13 should stay.

**Comment by fnielsen** [comment]
It is line 34 when it is not found.

**Pull request merged**